components/feature/autofill/src/main/java/mozilla/components/feature/autofill/handler/FillRequestHandler.kt

```diff
@@ -88,7 +72,7 @@ internal class FillRequestHandler(
      * [AssistStructure] is `null`
      */
     @Suppress("ReturnCount")
-    suspend fun handleConfirmation(structure: AssistStructure?, loginId: String): Dataset? {
+    suspend fun handleConfirmation(structure: RawStructure?, loginId: String): DatasetBuilder? {
```

Comment:
I think making RawStructure param as non-nullable would be a bit cleaner. Not much you can really do w/o a structure, so why allow it as `null`?

Reply:
The reason was for that check to be testable. If I make this non-nullable then this `null` check needs to go into `AbstractAutofillService` and there it cannot be tested.